### DISCUSSION TITLE
Combining 3 imagemagick cli parameters into one

### DISCUSSION THREAD

**Discussion by StealthRT** [ImageMagick/ImageMagick#4732] [Consulting]
Hey all I have these 3 imageMagick scripts (command line arguments) that I am trying to combine into one ImageMagick.NET code.

First (merging 2 images together):
```
convert ^
( testingl.jpg -resize 610x440^^ -gravity West -extent 1080x440 ) ^
( testingr.jpg -resize 610x440^^ -gravity East -extent 1080x440 ) ^
blend_mask.png -blur 0x7 ^
-composite bothImagesMerged.jpg
```
Second (Create 2 round objects with photo inside):
```
convert lisa.jpg -resize 100x100! ^
null: ( -size 100x100 xc:black -fill white -draw "circle 50,50 50,88" ) ^
-alpha off -compose copy_opacity -layers composite ^
null: ( -size 100x100 xc:"graya(100%,0)" -fill black -draw "circle 50,50 50,90" -blur 0x5 ) ^
-compose dstover -layers composite ^
-background none -gravity center +smush -25+0 ^
roundImageLisa.png

convert homer.jpg -resize 100x100! ^
null: ( -size 100x100 xc:black -fill white -draw "circle 50,50 50,88" ) ^
-alpha off -compose copy_opacity -layers composite ^
null: ( -size 100x100 xc:"graya(100%,0)" -fill black -draw "circle 50,50 50,90" -blur 0x5 ) ^
-compose dstover -layers composite ^
-background none -gravity center +smush -25+0 ^
roundImageHomer.png
```
Third (Write text on top of photo):
```
convert -size 1080x440 xc:none -gravity center ^
-font arial -pointsize 40 ^
-stroke black -strokewidth 2 -annotate +-330+-150 "Lisa Simpson" ^
-stroke black -strokewidth 2 -annotate +330+-150 "Homer Simpson" ^
-background none -shadow 520x3+0+0 +repage ^
-stroke none -fill white -annotate +-330+-150 "Lisa Simpson" ^
-stroke none -fill white -annotate +330+-150 "Homer Simpson" ^
bothImagesMerged.jpg +swap -gravity center -geometry +0-3 ^
-composite  textOverImg.jpg
```
If I was able to combine all 3 of those then the output would look something like this:

![image](https://user-images.githubusercontent.com/1037844/150186664-3ad846ac-94a0-4326-ab6a-3af8d1db46e6.png)

I've tried to put all of them into a one-liner but can not seem to find the correct way (order mainly) into doing so.

Assistance would be great!

**Comment by snibgo** [comment]
I'm not sure what you want help with: combining four commands into one, or converting them to .NET code.

I don't do .NET, so can't help there.

For combining into one: your final command makes one image, then reads the first, swaps them and composites them.

First step: make all "-compose" methods explicit. So your first and fourth commands would have "-compose Over". I would reverse the order of the images in the fourth command, putting the second entirely within parentheses, and removing "+swap".

So your fourth command starts by reading bothImagesMerged.jpg. Replace that by the contents of the first command, omitting the write to bothImagesMerged.jpg. Check that any settings carried from the first command won't cause problems in the remains of the fourth.

Your second and third commands have no relevance to each other or to the first and fourth. So there is little benefit from combining them. But you can if you want.

In general, suppose you have two commands:
```
convert {any_options} outputA.ext

convert {any_other_options} outputB.ext
```
You can combine them like this:
```
convert {any_options} -write outputA.ext -delete 0--1 {any_other_options} outputB.ext
```
Again, beware of conflicting settings, especially any that the second command assume have default values.
